packages/truffle-core/lib/debug/compiler.js

```diff
@@ -0,0 +1,34 @@
+const compile = require("truffle-compile");
+
+class DebugCompiler {
+  constructor(config) {
+    this.config = config;
+  }
+
+  async compile() {
+    const { contracts, files } = await new Promise((accept, reject) => {
```

Comment:
It is necessary to have the explicit Promise because `compile` passes _two_ success parameters to the callback, and in my observation, this fails to work through `promisify`

Reply:
I'll resolve this once I've added a comment explaining as such

Reply:
Oh huh.  Except that shouldn't apply in this case, right?  Because it only passes one (in that it wraps the two in an object).  Is there something I'm missing here?

Reply:
Cleaned this up a bit. Thanks for identifying the code smell!

Reply:
(It turned out I was missing something, yes.  But @gnidan and I decided to at least take as much outside of the manual promise as possible.)